### PR TITLE
Docker dashboard

### LINKED ISSUE #281: Dockerise and deploy the dashboard
As an amateur astronomer, I want the dashboard to be deployed online so I can easily access with a link, so that I can conveniently access all my star gazing information.

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #281 

What I have done:
- Dockerised dashboard
- Removed some files
- Fixed imports

### CODE REVIEW

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
-# Copy the current directory contents into the container at /app
-COPY . /app
+WORKDIR /app
 
-# Install any needed packages specified in requirements.txt
+COPY requirements.txt .
 RUN pip install --no-cache-dir -r requirements.txt
 
-# Expose the default port that Streamlit uses
+COPY aurora_map.py .
```
Comment (MurasatoNaoya) [inline dashboard/Dockerfile:11]: Good to copy the specific files needed, rather than the entire file directory.

File: dashboard/api_error.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        return self._message
+
+    @message.setter
+    def message(self, new_message):
+        '''Setter for the message property of the APIError class'''
+        self._message = f"{self._message}: {new_message}"
+
+    def define_error(self):
+        '''Returns the associated error message to the status code'''
+        error_messages = {
```
Comment (MurasatoNaoya) [inline dashboard/api_error.py:25]: Extensive list of error messages for API calls, for easier debugging. *unauthorised

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.

File: dashboard/Homepage.py [hunk 3]
```diff
@@ -1,21 +1,11 @@
...
 import pandas as pd
-import altair as alt
-from email_validator import validate_email as ve, EmailNotValidError
-import phonenumbers
-from phonenumbers import NumberParseException
-from load_dashboard_data import connect_to_db, load_from_starwatch_rds, load_forecasts_by_county_name, load_celestial_body_information
 from nasa_pipeline import nasa_pipeline, get_image_of_the_day, get_iss_location, get_moon_phase
-from aurora_map import create_aurora_map, create_visibility_map, create_body_visibility_map
-from azimuth_plot import make_sky_plot, get_bodies, get_star_chart, get_regions
+from aurora_map import create_aurora_map, create_visibility_map
```
Comment (CalebMathers) [inline dashboard/Homepage.py:8]: Good, no need to import the third one for this page

File: dashboard/extract.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+from dotenv import load_dotenv
+from psycopg2 import connect
+from requests import get
+from api_error import APIError
+
+URL = "https://api.open-meteo.com/v1/forecast"
+
+def get_connection():
+    '''Returns a psycopg2 connection'''
+    return connect(f"""dbname={ENV["DB_NAME"]} user={ENV["DB_USER"]}
```
Comment (CalebMathers) [inline dashboard/extract.py:13]: Do we need to load the dotenv here? Or could this function just be imported from another script?

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**